### PR TITLE
Auth integration tests for CSINodeInfo changes

### LINKED ISSUE #68254: Integration tests for CSINodeInfo NodeAuthorizer and NodeRestriction updates
/kind bug

A NodeAuthorizer and NodeRestriction update was introduced in https://github.com/kubernetes/kubernetes/pull/67684 to restrict node updates to `CSINodeInfo`. However, integration testing proved to be more difficult because the original test suite was not set up to include APIExtensions apiserver, so it wasn't possible to install CRDs. This issue tracks the effort for adding these integration tests [here](https://github.com/kubernetes/kubernetes/blob/master/test/integration/auth/node_test.go)

/assign
/cc @liggitt @saad-ali

### PR DESCRIPTION
**Pull request by verult**
**What this PR does / why we need it**: Adds integration tests for NodeAuthorizer and NodeRestriction changes for CSINodeInfo object.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubernetes/issues/68254

**Special notes for your reviewer**:
* The ServiceAccount admission plugin was retrying to fetch the missing "default" service account for ~1s per request, causing massive delays. Should the SA be pre-installed in the test server?
* Need to refactor to use CRD functions once #68579 merges.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fejta** [comment]
/uncc
Is wip

**Comment by verult** [comment]
/hold

Waiting for #68579 to merge, but otherwise ready for review

File: test/integration/auth/node_test.go [hunk 1]
```diff
@@ -87,36 +67,33 @@ func TestNodeAuthorizer(t *testing.T) {
...
-	versionedInformerFactory.Start(stopCh)
+	tokenFile.WriteString(strings.Join([]string{
+		fmt.Sprintf(`%s,admin,uid1,"system:masters"`, tokenMaster),
+		fmt.Sprintf(`%s,unknown,uid2,"system:nodes"`, tokenNodeUnknown),
+		fmt.Sprintf(`%s,system:node:node1,uid3,"system:nodes"`, tokenNode1),
+		fmt.Sprintf(`%s,system:node:node2,uid4,"system:nodes"`, tokenNode2),
+	}, "\n"))
+	tokenFile.Close()
+
+	server := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{
```
Comment (liggitt) [inline test/integration/auth/node_test.go]: clever

File: test/integration/auth/node_test.go [hunk 2]
```diff
@@ -175,6 +156,27 @@ func TestNodeAuthorizer(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	crd, err := superuserCRDClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(&apiextensionsv1beta1.CustomResourceDefinition{
```
Comment (liggitt) [inline test/integration/auth/node_test.go]: can this use the CRD defined under the csi-api package?

Comment (verult) [reply test/integration/auth/node_test.go]: Yup that's the plan, once the other PR merges

**Comment by verult** [comment]
Addressed comments

**Comment by verult** [comment]
/sig api-machinery
for API server integration review

/unhold

**Comment by liggitt** [comment]
/hold cancel
/lgtm

Thanks for following up on this

**Pull request merged**